engine/engine.go

```diff
@@ -71,6 +72,9 @@ type Options struct {
 	GitClient git.Git
 	// DockerOptions is used to customize docker operations.
 	DockerOptions docker.Options
+	// TimeoutSeconds is a limit to how long a docker image pull will last
+	// If no value is provided, it defaults to 3600
+	TimeoutSeconds int
```

Comment:
Is this only gonna be for docker pulls? Wouldn't we want this to be a general purpose timeout for all operations?

Reply:
i ended up using it for all operations as well